docs/user/reference/charm/index.md

```diff
@@ -41,24 +41,32 @@ Example machine charms:
 - [Vault](https://charmhub.io/vault)
 - [Rsyslog](https://charmhub.io/rsyslog)
 
+(infrastructure-agnostic-charm)=
+#### Infrastructure-agnostic
+
+While charms are still very much either for {ref}`Kubernetes <kubernetes-charm>` or {ref}`machines <machine-charm>`, some {ref}`workloadless <workloadless-charm>` charms are in fact infrastructure-agnostic and can deploy to both.
+
+```{note}
+That is because most of the difference between a machine charm and a Kubernetes charm comes from how the charm communicates with the workload (on machines they're in the same machine but on Kubernetes they are in different containers, even if on the same pod). So, if a charm does not have a workload, and its metadata does not stipulate Kubernetes, and the charm does not do anything that would only make sense on machines / Kubernetes, it can run perfectly fine on both machines and Kubernetes -- the details of the deployment will differ (the charm will be deployed on a machine vs. a container in a pod), but the deployment will be successful. Example workloadless charms that are cloud-agnostic: [Azure Storage Integrator](https://charmhub.io/azure-storage-integrator).
```

Comment:
> most of the difference between a machine charm and a Kubernetes charm comes from how the charm communicates with the workload

also: how it installs the workload. Many machine charms do `snap install`, whereas most k8s charms rely on juju to provision a container for the workload.

Reply:
Updated the language to "how a charm handles the workload".

Reply:
we could squint a bit and use "operate" which hopefully subsumes 'install+manage' :)

Reply:
Too late, merged. Will keep it in mind for future iterations!